### ISSUE TITLE
`generate` command fails if `.ioc` file has space in file name or directory

### ISSUE THREAD

**Issue by RossBencina** [ussserrr/stm32pio#21]
I have the following project directory:

  `C:\Users\Desktop\Ross\STM32Learning`

containing only the `.ioc` file named `STM32 Learning.ioc` (there is a space between "STM32" and "Learning".

I ran the following commands:

>C:\Python37\Scripts\stm32pio.exe init
>C:\Python37\Scripts\stm32pio.exe generate

Then the following error was displayed.

![image](https://user-images.githubusercontent.com/3253274/101758015-f4b3f900-3b2b-11eb-9475-367e191c6a4f.png)
 
The problem goes away if I rename the `.ioc` file without a space.

I get a similar error if the working directory has a space in the directory name, e.g. 

  `C:\Users\Desktop\Ross\STM32 Learning`

In the latter case, when running `generate` a new directory named  `C:\Users\Desktop\Ross\STM32` is created (e.g. the tool is ignoring the directory name from the space onwards).

I don't know the cause of the problem but possibly there are some missing quotation marks somewhere.

Please describe the bug and supplement the additional information:
 - Windows 10, Python 3.7.3
 - content of the config file `stm32pio.ini`:
```
[app]
java_cmd = java
platformio_cmd = platformio
cubemx_cmd = C:/Program Files/STMicroelectronics/STM32Cube/STM32CubeMX/STM32CubeMX.exe

[project]
cubemx_script_content = config load ${ioc_file_absolute_path}
        generate code ${project_dir_absolute_path}
        exit

platformio_ini_patch_content = [platformio]
        include_dir = Inc
        src_dir = Src

board =
ioc_file = STM32 Learning.ioc
```

 - your project folder tree (see above)

Verbose command output:

```
C:\Users\Ross\Desktop\STM32Learning>C:\Python37\Scripts\stm32pio.exe -v generate
DEBUG    setup_logging              debug logging enabled
DEBUG    _load_config               searching for stm32pio.ini...
DEBUG    _find_ioc_file             using 'STM32 Learning.ioc' file from the INI config
DEBUG    __init__                   resolved config:
========== app ==========
('java_cmd', 'java')
('platformio_cmd', 'platformio')
('cubemx_cmd', 'C:/Program Files/STMicroelectronics/STM32Cube/STM32CubeMX/STM32CubeMX.exe')

========== project ==========
('cubemx_script_content', 'config load ${ioc_file_absolute_path}\ngenerate code ${project_dir_absolute_path}\nexit')
('platformio_ini_patch_content', '[platformio]\ninclude_dir = Inc\nsrc_dir = Src')
('board', '')
('ioc_file', 'STM32 Learning.ioc')

INFO     generate_code              starting to generate a code from the CubeMX .ioc file...
log4j user configuration file not found: C:\Users\Ross/.stm32cubemx/log4j.properties
Configure log4j with default settings from jar:file:/C:/Program%20Files/STMicroelectronics/STM32Cube/STM32CubeMX/STM32CubeMX.exe!/log4j/log4j.stm32cubemx.properties
2020-12-10 21:15:41,923 [INFO] MicroXplorer:77 - [MX] MX Start == 2246992563080500
2020-12-10 21:15:41,926 [INFO] MicroXplorer:565 - Detected Java Version = 11.0.9.1
2020-12-10 21:15:41,933 [INFO] ApplicationProperties:181 - Using Application install path: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX
2020-12-10 21:15:41,949 [INFO] DbMcusXml:71 - Set database path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/mcu/
2020-12-10 21:15:41,956 [INFO] DbBoardsPdsc:57 - Set plugin database path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/boardmanager/
2020-12-10 21:15:41,957 [INFO] DbMcus:259 - Set plugin images path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/mcufinder/images/
2020-12-10 21:15:41,957 [INFO] DbBoards:265 - Set plugin images path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/mcufinder/images/
2020-12-10 21:15:41,960 [INFO] DbExamples:329 - Set plugin images path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/mcufinder/images/
2020-12-10 21:15:41,961 [INFO] McuFinderGlobals:40 - Set McuFinder mode to 1 (CubeMX integrated)
2020-12-10 21:15:41,969 [INFO] DbMcusDocs:112 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:41,971 [INFO] DbMcusJson:63 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:41,972 [INFO] DbBoardsDocs:112 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:41,974 [INFO] DbBoardsJson:56 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:42,245 [INFO] CrossReferenceDbSqlite:196 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/cs/
2020-12-10 21:15:42,249 [INFO] DbExamplesSqlite:699 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:42,474 [INFO] RulesReader:52 - Compatibility file has been processed (228 Rules)
2020-12-10 21:15:42,552 [INFO] STM32CubeMX:31 - Starting STM32CubeMX
2020-12-10 21:15:43,366 [INFO] MainPanel:190 - HeapMemory: 536870912
2020-12-10 21:15:43,535 [INFO] DbMcusXml:71 - Set database path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/mcu/
2020-12-10 21:15:43,535 [INFO] DbBoardsPdsc:57 - Set plugin database path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/boardmanager/
2020-12-10 21:15:43,535 [INFO] DbMcus:259 - Set plugin images path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/mcufinder/images/
2020-12-10 21:15:43,536 [INFO] DbBoards:265 - Set plugin images path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/mcufinder/images/
2020-12-10 21:15:43,536 [INFO] DbExamples:329 - Set plugin images path to: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\/plugins/mcufinder/images/
2020-12-10 21:15:43,537 [WARN] DbFile:36 - Overriding database path with different value: C:\Users\Ross\.stmcufinder\plugins\mcufinder/ => C:\Users\Ross\.stmcufinder\plugins\mcufinder
2020-12-10 21:15:43,537 [INFO] DbMcusDocs:112 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:43,537 [WARN] DbFile:36 - Overriding database path with different value: C:\Users\Ross\.stmcufinder\plugins\mcufinder/ => C:\Users\Ross\.stmcufinder\plugins\mcufinder
2020-12-10 21:15:43,538 [INFO] DbMcusJson:63 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:43,538 [WARN] DbFile:36 - Overriding database path with different value: C:\Users\Ross\.stmcufinder\plugins\mcufinder/ => C:\Users\Ross\.stmcufinder\plugins\mcufinder
2020-12-10 21:15:43,538 [INFO] DbBoardsDocs:112 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:43,538 [WARN] DbFile:36 - Overriding database path with different value: C:\Users\Ross\.stmcufinder\plugins\mcufinder/ => C:\Users\Ross\.stmcufinder\plugins\mcufinder
2020-12-10 21:15:43,539 [INFO] DbBoardsJson:56 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:43,539 [WARN] DbFile:36 - Overriding database path with different value: C:\Users\Ross\.stmcufinder\plugins\mcufinder/ => C:\Users\Ross\.stmcufinder\plugins\mcufinder
2020-12-10 21:15:43,539 [WARN] DbFile:36 - Overriding database path with different value: C:\Users\Ross\.stmcufinder\plugins\mcufinder/ => C:\Users\Ross\.stmcufinder\plugins\mcufinder
2020-12-10 21:15:43,539 [INFO] CrossReferenceDbSqlite:196 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/cs/
2020-12-10 21:15:43,540 [WARN] DbFile:36 - Overriding database path with different value: C:\Users\Ross\.stmcufinder\plugins\mcufinder/ => C:\Users\Ross\.stmcufinder\plugins\mcufinder
2020-12-10 21:15:43,540 [INFO] DbExamplesSqlite:699 - Set database path to: C:\Users\Ross\.stmcufinder\plugins\mcufinder//mcu/
2020-12-10 21:15:43,565 [INFO] ApplicationProperties:181 - Using Application install path: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX
2020-12-10 21:15:43,567 [INFO] PluginManage:178 - Search for loadable plugins [exclusion list=, ]
2020-12-10 21:15:43,569 [INFO] PluginManage:292 - Check plugin analytics
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.thoughtworks.xstream.core.util.Fields (file:C:/Program Files/STMicroelectronics/STM32Cube/STM32CubeMX/plugins/analytics.jar) to field java.util.TreeMap.comparator
WARNING: Please consider reporting this to the maintainers of com.thoughtworks.xstream.core.util.Fields
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2020-12-10 21:15:48,514 [INFO] AnalyticsPlugin:225 - Accepted Software Licenses:
2020-12-10 21:15:48,514 [INFO] AnalyticsPlugin:227 - Accepted CMSIS Pack Licenses:
2020-12-10 21:15:48,515 [INFO] AnalyticsPlugin:229 - Accepted Firmware Licenses:
2020-12-10 21:15:48,515 [INFO] PluginManage:342 - Loaded plugin analytics (category:tool,tabindex:-1)
2020-12-10 21:15:48,516 [INFO] PluginManage:292 - Check plugin clock
2020-12-10 21:15:48,527 [INFO] PluginManage:342 - Loaded plugin clock (category:base,tabindex:2)
2020-12-10 21:15:48,527 [INFO] PluginManage:292 - Check plugin ddr
2020-12-10 21:15:48,532 [INFO] PluginManage:342 - Loaded plugin ddr (category:tool,tabindex:6)
2020-12-10 21:15:48,532 [INFO] PluginManage:292 - Check plugin filemanager
2020-12-10 21:15:48,684 [INFO] PluginManage:342 - Loaded plugin filemanager (category:base,tabindex:10)
2020-12-10 21:15:48,685 [INFO] PluginManage:292 - Check plugin ipmanager
2020-12-10 21:15:48,698 [INFO] PluginManage:342 - Loaded plugin ipmanager (category:base,tabindex:5)
2020-12-10 21:15:48,699 [INFO] PluginManage:292 - Check plugin pinoutandconfiguration
2020-12-10 21:15:48,709 [INFO] PluginManage:342 - Loaded plugin pinoutandconfiguration (category:base,tabindex:1)
2020-12-10 21:15:48,709 [INFO] PluginManage:292 - Check plugin pinoutconfig
2020-12-10 21:15:48,770 [INFO] PluginManage:342 - Loaded plugin pinoutconfig (category:base,tabindex:0)
2020-12-10 21:15:48,771 [INFO] PluginManage:292 - Check plugin power
2020-12-10 21:15:48,785 [INFO] PluginManage:342 - Loaded plugin power (category:power,tabindex:4)
2020-12-10 21:15:48,785 [INFO] PluginManage:292 - Check plugin projectmanager
2020-12-10 21:15:48,864 [INFO] PluginManage:342 - Loaded plugin projectmanager (category:projectmanager,tabindex:3)
2020-12-10 21:15:48,865 [INFO] PluginManage:292 - Check plugin thirdparty
2020-12-10 21:15:49,027 [INFO] PluginManage:342 - Loaded plugin thirdparty (category:base,tabindex:-1)
2020-12-10 21:15:49,027 [WARN] IntegrityCheckThread:59 - waiting for thirdparty lock release [integrity check]
2020-12-10 21:15:49,027 [INFO] PluginManage:292 - Check plugin tools
2020-12-10 21:15:49,029 [INFO] IntegrityCheckThread:61 - entering critical section [integrity check]
2020-12-10 21:15:49,030 [INFO] ThirdPartyUpdaterWithRetryManager:56 - Updater plugin not ready yet. [1/15]
2020-12-10 21:15:49,032 [INFO] PluginManage:342 - Loaded plugin tools (category:base,tabindex:7)
2020-12-10 21:15:49,032 [INFO] PluginManage:292 - Check plugin tutovideos
2020-12-10 21:15:49,304 [INFO] PluginManage:342 - Loaded plugin tutovideos (category:base,tabindex:-1)
2020-12-10 21:15:49,305 [INFO] PluginManage:292 - Check plugin updater
2020-12-10 21:15:49,330 [INFO] PluginManage:342 - Loaded plugin updater (category:base,tabindex:12)
2020-12-10 21:15:49,331 [INFO] PluginManage:265 - PluginManage : Loaded plugins [13]
2020-12-10 21:15:49,570 [INFO] PinOutPanel:1434 - setPackage(No Configuration,No Configuration)
2020-12-10 21:15:49,675 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,675 [INFO] PluginManager:203 - loadIPPluginJar : add adc
2020-12-10 21:15:49,678 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,679 [INFO] PluginManager:203 - loadIPPluginJar : add aes
2020-12-10 21:15:49,681 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,682 [INFO] PluginManager:203 - loadIPPluginJar : add can
2020-12-10 21:15:49,684 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,684 [INFO] PluginManager:203 - loadIPPluginJar : add comp
2020-12-10 21:15:49,687 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,687 [INFO] PluginManager:203 - loadIPPluginJar : add cryp
2020-12-10 21:15:49,690 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,690 [INFO] PluginManager:203 - loadIPPluginJar : add dfsdm
2020-12-10 21:15:49,696 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,697 [INFO] PluginManager:203 - loadIPPluginJar : add dma
2020-12-10 21:15:49,699 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,699 [INFO] PluginManager:203 - loadIPPluginJar : add fatfs
2020-12-10 21:15:49,704 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,704 [INFO] PluginManager:203 - loadIPPluginJar : add fmc
2020-12-10 21:15:49,711 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,711 [INFO] PluginManager:203 - loadIPPluginJar : add freertos
2020-12-10 21:15:49,713 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,713 [INFO] PluginManager:203 - loadIPPluginJar : add genericplugin
2020-12-10 21:15:49,716 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,717 [INFO] PluginManager:203 - loadIPPluginJar : add gfxmmu
2020-12-10 21:15:49,723 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,723 [INFO] PluginManager:203 - loadIPPluginJar : add gic
2020-12-10 21:15:49,727 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,727 [INFO] PluginManager:203 - loadIPPluginJar : add gpio
2020-12-10 21:15:49,730 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,730 [INFO] PluginManager:203 - loadIPPluginJar : add gtzc
2020-12-10 21:15:49,733 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,733 [INFO] PluginManager:203 - loadIPPluginJar : add hash
2020-12-10 21:15:49,736 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,736 [INFO] PluginManager:203 - loadIPPluginJar : add i2c
2020-12-10 21:15:49,739 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,739 [INFO] PluginManager:203 - loadIPPluginJar : add i2s
2020-12-10 21:15:49,744 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,745 [INFO] PluginManager:203 - loadIPPluginJar : add ipddr
2020-12-10 21:15:49,747 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,748 [INFO] PluginManager:203 - loadIPPluginJar : add lorawan
2020-12-10 21:15:49,750 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,750 [INFO] PluginManager:203 - loadIPPluginJar : add ltdc
2020-12-10 21:15:49,754 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,754 [INFO] PluginManager:203 - loadIPPluginJar : add mdma
2020-12-10 21:15:49,758 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,758 [INFO] PluginManager:203 - loadIPPluginJar : add nvic
2020-12-10 21:15:49,761 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,761 [INFO] PluginManager:203 - loadIPPluginJar : add opamp
2020-12-10 21:15:49,764 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,765 [INFO] PluginManager:203 - loadIPPluginJar : add openamp
2020-12-10 21:15:49,768 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,768 [INFO] PluginManager:203 - loadIPPluginJar : add pdm2pcm
2020-12-10 21:15:49,774 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,777 [INFO] PluginManager:203 - loadIPPluginJar : add plateformsettings
2020-12-10 21:15:49,780 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,780 [INFO] PluginManager:203 - loadIPPluginJar : add quadspi
2020-12-10 21:15:49,784 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,784 [INFO] PluginManager:203 - loadIPPluginJar : add resmgrutility
2020-12-10 21:15:49,787 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,788 [INFO] PluginManager:203 - loadIPPluginJar : add sai
2020-12-10 21:15:49,790 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,790 [INFO] PluginManager:203 - loadIPPluginJar : add spi
2020-12-10 21:15:49,795 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,795 [INFO] PluginManager:203 - loadIPPluginJar : add stm32_wpan
2020-12-10 21:15:49,797 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,797 [INFO] PluginManager:203 - loadIPPluginJar : add tim
2020-12-10 21:15:49,801 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,801 [INFO] PluginManager:203 - loadIPPluginJar : add touchsensing
2020-12-10 21:15:49,804 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,804 [INFO] PluginManager:203 - loadIPPluginJar : add tracer_emb
2020-12-10 21:15:49,808 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,808 [INFO] PluginManager:203 - loadIPPluginJar : add ts
2020-12-10 21:15:49,810 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,811 [INFO] PluginManager:203 - loadIPPluginJar : add tsc
2020-12-10 21:15:49,813 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,813 [INFO] PluginManager:203 - loadIPPluginJar : add ucpd
2020-12-10 21:15:49,816 [INFO] IPUIPlugin:64 - create IPUIPlugin
2020-12-10 21:15:49,816 [INFO] PluginManager:203 - loadIPPluginJar : add usart
2020-12-10 21:15:49,846 [INFO] Updater:1002 - Updater Version found : 6.1.0
2020-12-10 21:15:49,866 [INFO] ApplicationProperties:181 - Using Application install path: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX
2020-12-10 21:15:50,020 [INFO] MainUpdater:2644 - connection check result : 10
2020-12-10 21:15:50,021 [INFO] MainUpdater:249 - Updater Check For Update Now.
2020-12-10 21:15:50,021 [INFO] MicroXplorer:415 - Change Database Version : DB.6.0.10
2020-12-10 21:15:50,083 [INFO] CheckServerUpdateThread:102 - End of CheckServer Thread
2020-12-10 21:15:50,471 [INFO] HomeAdPanel:51 - Auto-refresh data requested => check proxy status ongoing
2020-12-10 21:15:50,539 [INFO] LoadUrlFilesThread:162 - End of LoadServerUrlFiles without Thread
2020-12-10 21:15:50,551 [INFO] DetailPanel:344 - Set advertising image to C:/Users/Ross/STM32Cube/Repository/\stm32cubemx-promo_image.jpg
2020-12-10 21:15:50,556 [INFO] LoadUrlFilesThread:162 - End of LoadServerUrlFiles without Thread
2020-12-10 21:15:50,556 [INFO] DetailPanel:344 - Set advertising image to C:/Users/Ross/STM32Cube/Repository/\stm32cubemx-promo_image.jpg
2020-12-10 21:15:50,622 [INFO] MicroXplorer:385 - Change Database Path :
2020-12-10 21:15:50,623 [INFO] MicroXplorer:415 - Change Database Version : DB.6.0.10
2020-12-10 21:15:50,623 [WARN] ThirdParty:783 - waiting for thirdparty lock release [close project]
2020-12-10 21:15:52,073 [WARN] ConditionMgr:437 - getConditionDescription Invalid condition id : SIGFOX STM32 LIB Condition cause : null
2020-12-10 21:15:52,084 [WARN] ConditionMgr:437 - getConditionDescription Invalid condition id : Sigfox Third Party LIB Condition cause : null
2020-12-10 21:15:52,090 [WARN] ConditionMgr:1040 - genDependencies : Invalid condition id : SIGFOX STM32 LIB Condition cause : null
2020-12-10 21:15:52,090 [WARN] ConditionMgr:1040 - genDependencies : Invalid condition id : Sigfox Third Party LIB Condition cause : null
2020-12-10 21:15:52,091 [WARN] ConditionMgr:1040 - genDependencies : Invalid condition id : Sigfox Third Party LIB Condition cause : null
2020-12-10 21:15:52,092 [WARN] ConditionMgr:1040 - genDependencies : Invalid condition id : SIGFOX STM32 LIB Condition cause : null
2020-12-10 21:15:52,092 [WARN] ConditionMgr:1040 - genDependencies : Invalid condition id : SIGFOX STM32 LIB Condition cause : null
2020-12-10 21:15:52,092 [WARN] ConditionMgr:1040 - genDependencies : Invalid condition id : Sigfox Third Party LIB Condition cause : null
2020-12-10 21:15:52,476 [INFO] ThirdParty:887 - Integrity check success = true
2020-12-10 21:15:52,477 [INFO] IntegrityCheckThread:75 - exiting critical section [integrity check]
2020-12-10 21:15:52,477 [INFO] IntegrityCheckThread:78 - End integrity checks thread
2020-12-10 21:15:52,477 [INFO] ThirdParty:785 - entering critical section [close project]
2020-12-10 21:15:52,478 [INFO] ThirdParty:796 - exiting critical section [close project]
2020-12-10 21:15:52,481 [INFO] PinOutPanel:1434 - setPackage(No Configuration,No Configuration)
2020-12-10 21:15:52,566 [INFO] MainFrame:167 - [MX] HOME_PANEL == 2247003207398000
config load C:\Users\Ross\Desktop\STM32Learning\STM32 Learning.ioc
2020-12-10 21:15:52,586 [INFO] HomeAdPanel:149 - scale :0.9666048237476809
KO
generate code C:\Users\Ross\Desktop\STM32Learning
2020-12-10 21:16:00,557 [INFO] CodeGenerator:639 - code generatio: config db path: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\\db\
Exception in code generation java.lang.StringIndexOutOfBoundsException: begin 10, end 11, length 0
KO
exit
Bye bye
WARNING  generate_code              Undefined result from the CubeMX (neither error or success symptoms were found in the logs). Keep going but there might be an error
```

**Comment by ussserrr** [comment]
I can confirm it (also not only for Windows).

> I don't know the cause of the problem but possibly there are some missing quotation marks somewhere.

Yep, it's exactly like that. Turns out I manage paths correctly in Python but forgot to provide them properly to the CubeMX. Different software struggling with spaced names for years now so I made it a habit to never assign such names to my files/folders 😄  and so never test it either.

P.S. Probably I should reach the GitHub support again, I still don't receive any issues notifications at all...
